### PR TITLE
update jenkinsfile for cms deployments

### PR DESCRIPTION
**Pull request by tskinn**
## Description
This change is an attempt to make so that when the Jenkins pipeline runs more than once on the same commit it will assume that it is a content-only deploy and skip a few stages to cut down on the time it takes to get the content changes out.

It checks S3 for the git commit to know whether to just run a content deploy or regular build.

This relates to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/14607

## Acceptance criteria
- [ ] The automatic Jenkins job should run a normal build on the first run of a commit.
- [ ] Any consecutive builds should skip the lint and testing stages.

## Definition of done
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -252,33 +288,24 @@ node('vetsgov-general-purpose') {
...
-        build job: 'deploys/vets-website-staging', parameters: [
-          booleanParam(name: 'notify_slack', value: true),
-          stringParam(name: 'ref', value: commit),
-        ], wait: false
-        build job: 'deploys/vets-website-vagovstaging', parameters: [
-          booleanParam(name: 'notify_slack', value: true),
-          stringParam(name: 'ref', value: commit),
-        ], wait: false
+        if (!isCMSDeploy) {
+          runDeploy('deploys/vets-website-dev', commit)
```
Comment (ncksllvn) [inline Jenkinsfile]: Very nice helper!

**Review by ncksllvn — APPROVED** [review]
LGTM!

**Comment by ncksllvn** [comment]
@tskinn Would be awesome if can merge this by end of today. Just to make sure I'm not missing a detail, this combined with the Jenkinsfile at `vagov-content` will enable merges to `vagov-content` to publish to dev and staging. Is that accurate?

**Comment by tskinn** [comment]
@ncksllvn We need to pass a flag to the build command based on if its a cms deploy right? What is that?

**Comment by ncksllvn** [comment]
@tskinn Yes sir! It's `--content-deployment`. Example - `npm run build -- --content-deployment --brand-consolidation-enabled --buildtype=vagovstaging`

File: Jenkinsfile [hunk 2]
```diff
@@ -165,9 +200,14 @@ node('vetsgov-general-purpose') {
       for (int i=0; i<envNames.size(); i++) {
         def envName = envNames.get(i)
 
+        def buildCommand = "npm run build -- --content-deployment --buildtype=${envName}"
+        if (isCMSDeploy) {
+          buildCommand = "npm run build -- --content-deployment --brand-consolidation-enabled --buildtype=${envName}"
+        }
```
Comment (ncksllvn) [inline Jenkinsfile]: I think the `--content-deployment`  flag should be omitted by default, and only applied when `isCMSDeploy` is true. Maybe something like -

```groovy
def cmsDeployArg = ""
if (isCMSDeploy) {
    cmsDeployArg = "--content-deployment"
}
def buildCommand = "npm run build -- ${cmsDeployArg} --buildtype=${envName}"
```

**Pull request merged**